src/function/table/pragma_detailed_profiling_output.cpp

```diff
@@ -82,21 +103,42 @@ static void PragmaDetailedProfilingOutputFunction(ClientContext &context, const
                                                   DataChunk &output) {
 	auto &state = (PragmaDetailedProfilingOutputOperatorData &)*operator_state;
 	auto &data = (PragmaDetailedProfilingOutputData &)*bind_data_p;
+
 	if (!state.initialized) {
 		// create a ChunkCollection
 		auto collection = make_unique<ChunkCollection>();
 
+		// create a chunk
 		DataChunk chunk;
 		chunk.Initialize(data.types);
 
+		// Initialize ids
 		int operator_counter = 1;
+		int function_counter = 1;
+		int expression_counter = 1;
 		if (!context.query_profiler_history.GetPrevProfilers().empty()) {
+			// For each Operator
 			for (auto op : context.query_profiler_history.GetPrevProfilers().back().second.GetTreeMap()) {
-				int function_counter = 1;
-				if (op.second->info.has_executor) {
-					for (auto &info : op.second->info.executors_info->roots) {
-						ExtractExpressions(*collection, *info, chunk, operator_counter, function_counter,
-						                   op.second->info.executors_info->sample_tuples_count);
+				// For each Expression Executor
+				for (auto &ee : op.second->info.executors_info) {
+					// For each Expression tree
+					for (auto &et : ee.second->roots) {
+						SetValue(
+						    chunk, chunk.size(), operator_counter, "ExpressionRoot", expression_counter++,
+						    // Sometimes, cycle counter is not accurate, too big or too small. return 0 for those cases
+						    et->name,
+						    (et->time > 0 && et->time < (1UL << 63)) ? et->time / double(et->sample_tuples_count) : 0,
```

Comment:
What exactly does this computation do? Can we do this as a separate step instead of as a big ternary?

Reply:
Yes, just to make sure that the number of cycles is a valid number.